statsmodels/tsa/holtwinters.py

```diff
@@ -0,0 +1,726 @@
+            (alphac * (l[i - 1] + phi * b[i - 1]))
+        b[i] = (beta * (l[i] - l[i - 1])) + (betac * phi * b[i - 1])
+        s[i + m - 1] = (y_gamma[i - 1] / (l[i - 1] + phi *
+                                          b[i - 1])) + (gammac * s[i - 1])
+    return sqeuclidean((l + phi * b) * s[:-(m - 1)], y)
+
+#(M,M) (Md,M)
+
+
+def _holt_win_mul_mul_dam(x, xi, p, y, l, b, s, m, n, max_seen):
+    alpha, beta, gamma, phi, alphac, betac, gammac, y_alpha, y_gamma = _holt_win_init(
+        x, xi, p, y, l, b, s, m)
+    if alpha * beta == 0.0:
+        return max_seen
+    if beta > alpha or gamma > 1 - alpha:
+        return max_seen
+    for i in range(1, n):
+        l[i] = (y_alpha[i - 1] / s[i - 1]) + \
+            (alphac * (l[i - 1] * b[i - 1]**phi))
+        b[i] = (beta * (l[i] / l[i - 1])) + (betac * b[i - 1]**phi)
+        s[i + m - 1] = (y_gamma[i - 1] / (l[i - 1] *
+                                          b[i - 1]**phi)) + (gammac * s[i - 1])
+    return sqeuclidean((l * b**phi) * s[:-(m - 1)], y)
+
+#(A,A) (Ad,A)
+
+
+def _holt_win_add_add_dam(x, xi, p, y, l, b, s, m, n, max_seen):
+    alpha, beta, gamma, phi, alphac, betac, gammac, y_alpha, y_gamma = _holt_win_init(
+        x, xi, p, y, l, b, s, m)
+    if alpha * beta == 0.0:
+        return max_seen
+    if beta > alpha or gamma > 1 - alpha:
+        return max_seen
+    for i in range(1, n):
+        l[i] = (y_alpha[i - 1]) - (alpha * s[i - 1]) + \
+            (alphac * (l[i - 1] + phi * b[i - 1]))
+        b[i] = (beta * (l[i] - l[i - 1])) + (betac * phi * b[i - 1])
+        s[i + m - 1] = y_gamma[i - 1] - \
+            (gamma * (l[i - 1] + phi * b[i - 1])) + (gammac * s[i - 1])
+    return sqeuclidean((l + phi * b) + s[:-(m - 1)], y)
+
+#(M,A) (M,Ad)
+
+
+def _holt_win_mul_add_dam(x, xi, p, y, l, b, s, m, n, max_seen):
+    alpha, beta, gamma, phi, alphac, betac, gammac, y_alpha, y_gamma = _holt_win_init(
+        x, xi, p, y, l, b, s, m)
+    if alpha * beta == 0.0:
+        return max_seen
+    if beta > alpha or gamma > 1 - alpha:
+        return max_seen
+    for i in range(1, n):
+        l[i] = (y_alpha[i - 1]) - (alpha * s[i - 1]) + \
+            (alphac * (l[i - 1] * b[i - 1]**phi))
+        b[i] = (beta * (l[i] / l[i - 1])) + (betac * b[i - 1]**phi)
+        s[i + m - 1] = y_gamma[i - 1] - \
+            (gamma * (l[i - 1] * b[i - 1]**phi)) + (gammac * s[i - 1])
+    return sqeuclidean((l * phi * b) + s[:-(m - 1)], y)
+
+
+class HoltWintersResults(Results):
+    def __init__(self, model, params, **kwds):
+        self.data = model.data
+        super(HoltWintersResults, self).__init__(model, params, **kwds)
+
+    def predict(self, start=None, end=None):
+        return self.model.predict(self.params, start, end)
+
+    def forecast(self, steps=1):
+        try:
+            start = self.model._index[-1] + 1
+            end = self.model._index[-1] + steps
+            return self.model.predict(self.params, start=start, end=end)
+        except:
```

Comment:
What's the specific exception that this captures?
We never use a unspecified except, since it hides everything.

Reply:
done